docs/sdks/rust/quickstart.md

```diff
@@ -78,123 +80,133 @@ use module_bindings::*;
 
 To `client/src/main.rs`, add:
 
 ```rust
-use spacetimedb_sdk::{
-    Address,
-    disconnect,
-    identity::{load_credentials, once_on_connect, save_credentials, Credentials, Identity},
-    on_disconnect, on_subscription_applied,
-    reducer::Status,
-    subscribe,
-    table::{TableType, TableWithPrimaryKey},
-};
+use spacetimedb_sdk::{anyhow, DbContext, Event, Identity, Status, Table, TableWithPrimaryKey};
+use spacetimedb_sdk::credentials::File;
 ```
 
-## Define main function
+## Define the main function
 
-We'll work outside-in, first defining our `main` function at a high level, then implementing each behavior it needs. We need `main` to do five things:
+Our `main` function will do the following:
+1. Connect to the database. This will also start a new thread for handling network messages.
+2. Handle user input from the command line.
 
-1. Register callbacks on any events we want to handle. These will print to standard output messages received from the database and updates about users' names and online statuses.
-2. Establish a connection to the database. This will involve authenticating with our credentials, if we're a returning user.
-3. Subscribe to receive updates on tables.
-4. Loop, processing user input from standard input. This will be how we enable users to set their names and send messages.
-5. Close our connection. This one is easy; we just call `spacetimedb_sdk::disconnect`.
-
-To `client/src/main.rs`, add:
+We'll see the implementation of these functions a bit later, but for now add to `client/src/main.rs`:
 
 ```rust
 fn main() {
-    register_callbacks();
-    connect_to_db();
-    subscribe_to_tables();
-    user_input_loop();
+    // Connect to the database
+    let conn = connect_to_db();
+    // Handle CLI input
+    user_input_loop(&conn);
 }
 ```
 
+
 ## Register callbacks
 
 We need to handle several sorts of events:
 
 1. When we connect and receive our credentials, we'll save them to a file so that the next time we connect, we can re-authenticate as the same user.
-2. When a new user joins, we'll print a message introducing them.
-3. When a user is updated, we'll print their new name, or declare their new online status.
-4. When we receive a new message, we'll print it.
-5. When we're informed of the backlog of past messages, we'll sort them and print them in order.
-6. If the server rejects our attempt to set our name, we'll print an error.
-7. If the server rejects a message we send, we'll print an error.
+2. When a new user joins, we'll print a message introducing them.	
+3. When a user is updated, we'll print their new name, or declare their new online status.	
+4. When we receive a new message, we'll print it.	
+5. When we're informed of the backlog of past messages, we'll sort them and print them in order.	
+6. If the server rejects our attempt to set our name, we'll print an error.	
+7. If the server rejects a message we send, we'll print an error.	
```

Comment:
Can you do your editor's equivalent of `M-x delete-trailing-whitespace`? There's several instances of trailing whitespace in this file.